src/test/java/teammates/test/cases/webapi/GetCourseSectionNamesActionTest.java

```diff
@@ -0,0 +1,86 @@
+import teammates.common.exception.EntityNotFoundException;
+import teammates.common.util.Const;
+import teammates.ui.webapi.action.GetCourseSectionNamesAction;
+import teammates.ui.webapi.action.JsonResult;
+import teammates.ui.webapi.output.CourseSectionNamesData;
+
+/**
+ * SUT: {@link GetCourseSectionNamesAction}.
+ */
+public class GetCourseSectionNamesActionTest extends BaseActionTest<GetCourseSectionNamesAction> {
+
+    @Override
+    protected String getActionUri() {
+        return Const.ResourceURIs.COURSE_SECTIONS;
+    }
+
+    @Override
+    protected String getRequestMethod() {
+        return GET;
+    }
+
+    @Test
+    @Override
+    protected void testExecute() throws Exception {
+        //See test cases below
+    }
+
+    @Test
+    protected void testExecute_typicalUsage_shouldPass() throws Exception {
+        InstructorAttributes instructor1OfCourse1 = typicalBundle.instructors.get("instructor1OfCourse1");
+        List<String> expectedSectionNames = logic.getSectionNamesForCourse(instructor1OfCourse1.getCourseId());
+
+        loginAsInstructor(instructor1OfCourse1.googleId);
+
+        ______TS("typical success case for instructor");
+
+        String[] params = {
+                Const.ParamsNames.COURSE_ID, instructor1OfCourse1.getCourseId(),
+        };
+        GetCourseSectionNamesAction getCourseSectionNamesAction = getAction(params);
+        JsonResult response = getJsonResult(getCourseSectionNamesAction);
+
+        assertEquals(HttpStatus.SC_OK, response.getStatusCode());
+
+        CourseSectionNamesData courseSectionNamesData = (CourseSectionNamesData) response.getOutput();
+        assertEquals(expectedSectionNames, courseSectionNamesData.getSectionNames());
+    }
+
+    @Test
+    protected void testExecute_nonExistCourse_shouldFail() throws Exception {
+        InstructorAttributes instructor1OfCourse1 = typicalBundle.instructors.get("instructor1OfCourse1");
+        loginAsInstructor(instructor1OfCourse1.googleId);
+
+        ______TS("failed case for non-existent course");
+
+        String[] params = {
+                Const.ParamsNames.COURSE_ID, "dummy-course",
+        };
+        assertThrows(EntityDoesNotExistException.class, () -> logic.getSectionNamesForCourse("dummy-course"));
+
+        GetCourseSectionNamesAction getCourseSectionNamesAction = getAction(params);
+        assertThrows(EntityNotFoundException.class, () -> getJsonResult(getCourseSectionNamesAction));
+    }
+
+    @Test
+    @Override
+    protected void testAccessControl() throws Exception {
+        InstructorAttributes instructor1OfCourse1 = typicalBundle.instructors.get("instructor1OfCourse1");
+
+        String[] submissionParams = new String[] {
+                Const.ParamsNames.COURSE_ID, instructor1OfCourse1.getCourseId(),
+                Const.ParamsNames.ENTITY_TYPE, Const.EntityType.INSTRUCTOR,
+        };
+
+        verifyOnlyInstructorsOfTheSameCourseCanAccess(submissionParams);
```

Comment:
Shouldn't we add other verifications like `verifyInaccessibleWithoutLogin`?

You can check examples in other test cases.

Reply:
Additional verifications are added.